test/app.e2e-spec.ts

```diff
@@ -55,6 +60,19 @@ describe('AppController (e2e)', () => {
     expect(onEventPatternCall).toBeDefined();
     expect(onEventPatternCall.payload).toEqual({ eventId, foo: 'bar' });
   });
+  it('should be able to catch and re-queue', async () => {
+    const topic = 'topic01';
+    await nsqd.createTopic('topic01'); // otherwise it fails on the 1st time as topic does not exist
+    await setTimeout(1500); // wait 1.5s for consumer polls the topic
+    const eventId = uuid() + 'thrown';
+    await nsqd.publish(topic, { eventId, foo: 'bar' });
+    let discardCalled = mockDiscardHdlr.mock.calls.length;
+    while (discardCalled === 0) {
+      await setTimeout(500);
+      discardCalled = mockDiscardHdlr.mock.calls.length;
```

Comment:
mockDiscardHdlr is a global mockFn, should call `mockClear` before/after each case

Reply:
fixed.